### PR TITLE
Upgrade raft library

### PR DESCRIPTION
**Pull request by briankassouf**

### CODE REVIEW

File: physical/raft/snapshot_test.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func TestRaft_Snapshot_Index(t *testing.T) {
 
 	// Get index
 	index, _ = raft.fsm.LatestState()
-	if index.Term != 1 {
+	if index.Term != 2 {
 		t.Fatalf("unexpected term, got %d expected 1", index.Term)
```
Comment (calvn) [inline physical/raft/snapshot_test.go:172]:
```suggestion
		t.Fatalf("unexpected term, got %d expected 2", index.Term)
```

File: physical/raft/snapshot_test.go [hunk 2]
```diff
@@ -168,7 +168,7 @@ func TestRaft_Snapshot_Index(t *testing.T) {
 
 	// Get index
 	index, _ = raft.fsm.LatestState()
-	if index.Term != 1 {
+	if index.Term != 2 {
```
Comment (calvn) [inline physical/raft/snapshot_test.go:171]: How come the term is now 2?

Comment (briankassouf) [reply physical/raft/snapshot_test.go:171]: Change upstream in the raft library, we used to rely on a `StartAsLeader` flag that was removed. Now the voting takes place an increases the term number

File: physical/raft/snapshot_test.go [hunk 3]
```diff
@@ -148,7 +148,7 @@ func TestRaft_Snapshot_Index(t *testing.T) {
 
 	// Get index
 	index, _ := raft.fsm.LatestState()
-	if index.Term != 1 {
+	if index.Term != 2 {
 		t.Fatalf("unexpected term, got %d expected 1", index.Term)
```
Comment (briankassouf) [inline physical/raft/snapshot_test.go:152]:
```suggestion
		t.Fatalf("unexpected term, got %d expected 2", index.Term)
```

**Pull request merged**